app/models/job.rb

```diff
@@ -36,6 +47,18 @@ def self.current_job_timeout(_timeout_adjustment = 1)
 
   delegate :current_job_timeout, :to => :class
 
+  def update_linked_task
+    return if miq_task.nil?
+    attributes = {}
+    attributes[:context_data] = context if previous_changes['context']
+    attributes[:started_on] = started_on if previous_changes['started_on']
+    attributes[:zone] = zone if previous_changes['zone']
+    attributes[:state] = state.try(:capitalize) if previous_changes['state']
+    attributes[:status] = status.try(:capitalize) if previous_changes['status']
+    attributes[:message] = message.truncate(255) if previous_changes['message']
+    miq_task.update_attributes!(attributes)
```

Comment:
I suggest writing `task_attributes` that has the `job.create_miq_task` values.

Then here use `miq_task.update_attributes(task_attributes)`